src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/ProvenBlockHeaderStore.cs

```diff
@@ -205,9 +205,30 @@ public void AddToPendingBatch(ProvenBlockHeader provenBlockHeader, HashHeightPai
 
             lock (this.lockObject)
             {
+                if (this.pendingTipHashHeight != null && provenBlockHeader.HashPrevBlock != this.pendingTipHashHeight.Hash)
+                {
+                    // The latest proven header to be added is the most recent from consensus
+                    // and is always assumed to be the consensus tip.
+                    // If a non-consecutive item is added then there may have been a reorg in the chain
+                    // this can happen when the node is in the middle of rewinding its consensus
```

Comment:
This line is incorrect. This can happen after rewind already happened and we pushed at least 1 block after the forkpoint

Reply:
hmm ah yeah, though I consider the entire switch from one chain to another as the rewind process
I will fix it to be `this can happen after the node completed a rewind of its consensus`